ebs/error_budget.py

```
@@ -417,6 +410,407 @@ def run_etc(self, wfe, wfsc_factor, sensitivity
+
+        Reference
+        ---------
+        - See <Post-Processing Factor> document for mathematical description
+
+        """
+        if (self.wfe is not None and self.wfsc_factor is not None 
+            and self.sensitivity is not None and self.contrast is not None):
+            self.post_wfsc_wfe = np.multiply(self.wfe, self.wfsc_factor)
+            delta_contrast = np.empty(self.contrast.shape[0])
+            for n in range(len(delta_contrast)):
+                delta_contrast[n] = np.sqrt((np.multiply(self.sensitivity[n]
+                                         , self.post_wfsc_wfe)**2).sum()
+                                           ) 
+            return 1E-12*delta_contrast
+        else: 
+            print("Need to assign wfe, wfsc_factor, sensitivity, " + 
+                  "and contrast values before determining delta_contrast") 
+
+    @property
+    def ppFact(self):
+        """
+        Compute the post-processing factor and assign the array to 
+        `self.ppFact`. 
+
+        Reference
+        ---------
+        - See <Post-Processing Factor> document for mathematical description
+
+        """
+        ppFact = self.delta_contrast/self.contrast
+        return np.where(ppFact>1.0, 1.0, ppFact)
+
+    
+    def write_ppFact_fits(self):
+        """
+        Create FITS file of ppFact array with randomized filename.  
+
+        """
+        input_dir = self.config['paths']['input']
+        if self.angles is not None:
+            random_string = str(int(1e10*np.random.rand()))
+            filename = "ppFact_"+random_string+".fits"
+            path = os.path.join(input_dir, filename)
+            with open(path, 'wb') as f:
+                arr = np.vstack((self.angles, self.ppFact)).T
+                fits.writeto(f, arr, overwrite=True)
+                self.ppFact_filename = path
+            self.trash_can.append(path)
+        else:  
+            print("Need to assign angle values to write ppFact FITS file")
```

same here as before, I would maybe raise an error